### DISCUSSION TITLE
Pattern to always update field on insert/update (e.g. with user id)

### DISCUSSION THREAD

**Discussion by tgross35** [diesel-rs/diesel#3565] [Q&A] [answered]
Is there a good pattern to require a field be updated whenever a model is updated?

My use case is having a column `updated_by_user_id` that is present on almost every model, and needs to be set with every update statement. I'm used to using something like Sqlalchemy's [`onupdate`](https://docs.sqlalchemy.org/en/20/core/defaults.html#python-executed-functions) argument, which could grab the user ID from a global function.

Is there a good way to mimic this behavior? I can't think of a good pattern that would require a field to be `.set()` every time, outside of a wrapper function that might not give the same configurability.

**Comment by weiznich** [comment, marked as answer]
The correct way to manage something like that is by using a sql trigger. Diesel even provides a simple function to register such triggers in your migration for a column called `updated_at`: https://github.com/diesel-rs/diesel/blob/master/diesel_cli/src/setup_sql/postgres/initial_setup/up.sql#L8

**Reply by tgross35** [reply]
Thank you!
